### ISSUE TITLE
On JRuby, removing a namespaced attribute doesn't work

### ISSUE THREAD

**Issue by alex** [sparklemotion/nokogiri#1299]
This code reproduces the issue:

```ruby
#!/usr/bin/env jruby

require 'nokogiri'

puts RUBY_ENGINE
doc = Nokogiri::XML(<<-EOF)
<?xml version="1.0" encoding="UTF-8"?>
<node xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
    <otherNode soapenv:mustUnderstand="1"></otherNode>
</node>
EOF

node = doc.at_xpath("//otherNode")
puts node.attributes
puts node.remove_attribute("mustUnderstand")
puts doc.to_s
```

When run under CRuby, `remove_attribute` will return the value `"1"`, and `doc.to_s` will show the XML without the attribute. On JRuby `remove_attribute` returns `nil`, and `doc.to_s` shows that `mustUnderstand` is still present.

**Comment by alex** [comment]
Digging in a bit more, the problem appears to be with `has_attribute?`, if I use `node.attributes["mustUnderstand"].remove`, the code works fine.

**Comment by alex** [comment]
Reading between the lines on http://www.w3.org/2003/01/dom2-javadoc/org/w3c/dom/Element.html#hasAttribute_java.lang.String_, it looks like `Element.hasAttribute` doesn't support namespaced elements, and a caller must use `Element. hasAttributeNS`.

**Comment by yokolet** [comment]
Hi, thanks for reporting!
Since this is not a complicated bug, I fixed the bug in the commit, 1be8831 . As far as I tested, the result was the same as CRuby version.

**Comment by yokolet** [comment]
Aw, actually, the commit, e1b5e34518cfc2c2f601c6e3d7723333bbb29f94 .

**Comment by alex** [comment]
Awesome! Thank you!

**Comment by flavorjones** [comment]
Sorry, I need to revert this, as it's not apparent what the meaning of this code would be given the current implementations in both JRuby and C:

```
        doc = Nokogiri::XML <<-EOXML
          <?xml version="1.0" encoding="UTF-8"?>
          <root xmlns:foo="http://foo.xmlsoap.org/"
                xmlns:bar="http://bar.xmlsoap.org/">
            <node foo:myAttr="foo" bar:myAttr="bar"></node>
          </root>
        EOXML

        doc.at_xpath("//node").remove_attribute("myAttr")
```

And I think the right thing to do is to add a new method, analogous to `Node#attribute_with_ns`, which would remove an attribute by specifying the namespace: `Node#remove_attribute_with_ns`.

Further, I think we could reasonably allow an argument to `#attribute` and `#remove_attribute` that would allow specifying the namespace (and thereby deprecate the `_with_ns` flavors in a future release).

**Comment by flavorjones** [comment]
@alex as a workaround in the meantime, you can find the appropriate member of `Node#attribute_nodes` and call `#remove` on it.

**Comment by alex** [comment]
@flavorjones Yup, that's what I'm doing `node.attributes["mustUnderstand"].remove` works fine

**Comment by flavorjones** [comment]
@alex well, again, that *happens* to work because you don't have any name collisions. In the example I give above, you would perhaps not get the results you wanted. I'd recommend using `Node#attributes_with_ns` if there's any possible question.

**Comment by alex** [comment]
Thanks.

**Comment by flavorjones** [comment]
The action item here is: write a method `Node#remove_attribute_with_ns`
